crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -1019,6 +1040,126 @@ impl IdentityManager {
             _ => Ok(None),
         }
     }
+
+    /// Given a list of changed devices, update any [`InboundGroupSession`]s
+    /// which were sent from those devices and which do not have complete
+    /// sender data.
+    ///
+    /// Returns a list of modified InboundGroupSessions which need storing in
+    /// the database.
+    async fn update_sender_data_from_device_changes(
+        &self,
+        device_changes: &DeviceChanges,
+    ) -> Result<Vec<InboundGroupSession>, CryptoStoreError> {
+        let mut result = Vec::new();
+
+        for device in device_changes.new.iter().chain(device_changes.changed.iter()) {
+            // 1. Look for InboundGroupSessions from the device whose sender_data is
+            //    UnknownDevice. For such sessions, we now have the device, and can update
+            //    the sender_data accordingly.
+            //
+            // In theory, we only need to do this for new devices. In practice, I'm a bit
+            // worried about races leading us to getting stuck in the
+            // UnknownDevice state, so we'll paper over that by doing this check
+            // on device updates too.
+            let mut result1 = self
```

Comment:
`result1` -> `sessions_unknown` or similar?

Reply:
removed as a side-effect of 74e63219d